src/compiler/factory/nodeFactory.ts

```diff
@@ -1184,7 +1182,7 @@ export function createNodeFactory(flags: NodeFactoryFlags, baseFactory: BaseNode
             const array = elements.slice() as MutableNodeArray<T>;
             array.pos = elements.pos;
             array.end = elements.end;
-            array.hasTrailingComma = hasTrailingComma;
+            array.hasTrailingComma = hasTrailingComma || elements.hasTrailingComma;

```

Comment:
`||` is probably not ideal since someone can call this with an explicit `false`.

```suggestion
            array.hasTrailingComma = hasTrailingComma ?? elements.hasTrailingComma;
```